src/main/java/org/alfresco/repo/solr/SOLRAdminClient.java

```diff
@@ -147,76 +122,176 @@ public void setScheduler(Scheduler scheduler)
     {
         this.scheduler = scheduler;
     }
+    
+    /**
+     * SOLR properties identified by store like "alfresco" or "archive"
+     * @param storeMappings
+     */
+    public void setStoreMappings(List<SolrStoreMapping> storeMappings) {
+        this.storeMappings = storeMappings;
+    }
+    
+    /* (non-Javadoc)
+     * @see org.springframework.beans.factory.BeanFactoryAware#setBeanFactory(org.springframework.beans.factory.BeanFactory)
+     */
+    @Override
+    public void setBeanFactory(BeanFactory beanFactory) throws BeansException
+    {
+        this.beanFactory = beanFactory;
+    }
+    
+    public void setShardRegistry(ShardRegistry shardRegistry) {
+        this.shardRegistry = shardRegistry;
+    }
+    
+    public void setUseDynamicShardRegistration(boolean useDynamicShardRegistration) {
+        this.useDynamicShardRegistration = useDynamicShardRegistration;
+    }
+
+    /* (non-Javadoc)
+     * @see org.springframework.beans.factory.InitializingBean#afterPropertiesSet()
+     */
+    @Override
+    public void afterPropertiesSet() throws Exception
+    {
+        mappingLookup.clear();
+        for(SolrStoreMapping mapping : storeMappings)
+        {
+            mappingLookup.put(mapping.getStoreRef(), new ExplicitSolrStoreMappingWrapper(mapping, beanFactory));
+        }
+    }
 
     public void init()
 	{
-    	ParameterCheck.mandatory("solrHost", solrHost);
-    	ParameterCheck.mandatory("solrPort", solrPort);
-    	ParameterCheck.mandatory("solrUser", solrUser);
-    	ParameterCheck.mandatory("solrPassword", solrPassword);
-    	ParameterCheck.mandatory("solrPingCronExpression", solrPingCronExpression);
-    	ParameterCheck.mandatory("solrConnectTimeout", solrConnectTimeout);
-
-		try
-		{
-	    	StringBuilder sb = new StringBuilder();
-	    	sb.append(httpClientFactory.isSSL() ? "https://" : "http://");
-	    	sb.append(solrHost);
-	    	sb.append(":");
-	    	sb.append(httpClientFactory.isSSL() ? solrSSLPort: solrPort);
-	    	sb.append(baseUrl);
-			this.solrUrl = sb.toString();
-			HttpClient httpClient = httpClientFactory.getHttpClient();
+		this.solrTracker = new SolrTracker(scheduler);
+	}

```

Comment:
Mixed tabs and spaces?  (I think it was mixed before you started)

Reply:
This is removed code.